unit_test/test_spdm_responder/csr.c

```diff
@@ -1333,6 +1336,519 @@ void libspdm_test_responder_csr_case12(void **state)
+    assert_int_equal(spdm_response->header.param2, 0);
+
+    /*matched csr_tracking_tag without overwrite*/
+    csr_tracking_tag = 1;
+    m_libspdm_get_csr_request->header.spdm_version = SPDM_MESSAGE_VERSION_13;
+    m_libspdm_get_csr_request->header.request_response_code = SPDM_GET_CSR;
+    m_libspdm_get_csr_request->header.param1 = 0;
+    m_libspdm_get_csr_request->header.param2 =
+        csr_tracking_tag << SPDM_GET_CSR_REQUEST_ATTRIBUTES_CSR_TRACKING_TAG_OFFSET;
+
+    m_libspdm_get_csr_request->opaque_data_length = 0;
+    m_libspdm_get_csr_request->requester_info_length = req_info_len;
+    libspdm_copy_mem(m_libspdm_get_csr_request + 1, req_info_len,
+                     right_req_info, req_info_len);
+
+    response_size = sizeof(response);
+    status = libspdm_get_response_csr(spdm_context,
+                                      m_libspdm_get_csr_request_size,
+                                      m_libspdm_get_csr_request,
+                                      &response_size, response);
+    /*second get_csr after device reset: get the responder cached csr*/
+    assert_int_equal(status, LIBSPDM_STATUS_SUCCESS);
+    assert_int_equal(response_size, sizeof(spdm_error_response_t));
+    spdm_response = (void *)response;
+    assert_int_equal(spdm_response->header.request_response_code,
+                     SPDM_ERROR);
+    assert_int_equal(spdm_response->header.param1,
+                     SPDM_ERROR_CODE_BUSY);
+    assert_int_equal(spdm_response->header.param2, 0);
+
+
+    /*matched csr_tracking_tag with overwrite*/
+    csr_tracking_tag = 1;
+    m_libspdm_get_csr_request->header.spdm_version = SPDM_MESSAGE_VERSION_13;
+    m_libspdm_get_csr_request->header.request_response_code = SPDM_GET_CSR;
+    m_libspdm_get_csr_request->header.param1 = 0;
+    m_libspdm_get_csr_request->header.param2 =
+        (csr_tracking_tag << SPDM_GET_CSR_REQUEST_ATTRIBUTES_CSR_TRACKING_TAG_OFFSET) |
+        SPDM_GET_CSR_REQUEST_ATTRIBUTES_OVERWRITE;
+
+    m_libspdm_get_csr_request->opaque_data_length = 0;
+    m_libspdm_get_csr_request->requester_info_length = req_info_len;
+    libspdm_copy_mem(m_libspdm_get_csr_request + 1, req_info_len,
+                     right_req_info, req_info_len);
+
+    response_size = sizeof(response);
+    status = libspdm_get_response_csr(spdm_context,
+                                      m_libspdm_get_csr_request_size,
+                                      m_libspdm_get_csr_request,
+                                      &response_size, response);
+    /*second get_csr after device reset: get the responder cached csr*/
+    assert_int_equal(status, LIBSPDM_STATUS_SUCCESS);
+    assert_int_equal(response_size, sizeof(spdm_error_response_t));
+    spdm_response = (void *)response;
+    assert_int_equal(spdm_response->header.request_response_code,
+                     SPDM_ERROR);
+    assert_int_equal(spdm_response->header.param1,
+                     SPDM_ERROR_CODE_UNEXPECTED_REQUEST);

```

Comment:
I think here `SPDM_ERROR_CODE_UNEXPECTED_REQUEST` should be `SPDM_ERROR_CODE_INVALID_REQUEST`.
Based on the  https://github.com/DMTF/libspdm/issues/2291#issuecomment-1822340186
![image](https://github.com/DMTF/libspdm/assets/90754803/264ff54d-3e38-4fce-a69d-05ec0214253a)

Reply:
Thanks. I have changed the returned error code.